editorconfig.el

```diff
@@ -589,6 +589,17 @@ This function will revert buffer when the coding-system has been changed."
               (revert-buffer-with-coding-system coding-system)))
         (setq editorconfig--apply-coding-system-currently nil)))))
 
+(defun editorconfig--delete-final-newline ()
+  "Ensure the buffer does not end with a newline.
+Does nothing if the buffer is read-only, of course."
+  (unless buffer-read-only
+    (save-excursion
+      (save-restriction
+        (widen)
+        (goto-char (point-max))
+        (while (looking-at-p "^$")
+          (delete-char -1))))))
```

Comment:
Does this work with CRLF ending?

Reply:
Yes, `^`, `$`, etc. would all be aware of the newline settings for the buffer.

Here are the relevant entries from info:

```
‘^’
     When matching a buffer, ‘^’ matches the empty string, but only at
     the beginning of a line in the text being matched (or the beginning
     of the accessible portion of the buffer).  Otherwise it fails to
     match anything.  Thus, ‘^foo’ matches a ‘foo’ that occurs at the
     beginning of a line.

     When matching a string instead of a buffer, ‘^’ matches at the
     beginning of the string or after a newline character.

     For historical compatibility, ‘^’ is special only at the beginning
     of the regular expression, or after ‘\(’, ‘\(?:’ or ‘\|’.  Although
     ‘^’ is an ordinary character in other contexts, it is good practice
     to use ‘\^’ even then.

‘$’
     is similar to ‘^’ but matches only at the end of a line (or the end
     of the accessible portion of the buffer).  Thus, ‘x+$’ matches a
     string of one ‘x’ or more at the end of a line.

     When matching a string instead of a buffer, ‘$’ matches at the end
     of the string or before a newline character.

     For historical compatibility, ‘$’ is special only at the end of the
     regular expression, or before ‘\)’ or ‘\|’.  Although ‘$’ is an
     ordinary character in other contexts, it is good practice to use
     ‘\$’ even then.
```